src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRanger.java

```diff
@@ -553,21 +553,33 @@ private double getAimHeight()
     @Override
     public void moveInAttackPosition()
     {
-        if (buildingGuards.getTask() == GuardTask.GUARD)
-        {
-            ((MinecoloniesAdvancedPathNavigate) worker.getNavigator()).setPathJob(new PathJobCanSee(worker, target, world, buildingGuards.getGuardPos(), 20),
-              null,
-              getCombatMovementSpeed());
-            return;
-        }
+        ((MinecoloniesAdvancedPathNavigate) worker.getNavigator()).setPathJob(new PathJobCanSee(worker,
+            target,
+            world,
+            buildingGuards.getGuardPos(), shouldStayCloseToPos()
+             ? 20 : 40),
+          null,
+          getCombatMovementSpeed());
+    }
 
-        worker.getNavigator()
-          .tryMoveToBlockPos(worker.getPosition().offset(BlockPosUtil.getXZFacing(target.getPosition(), worker.getPosition()).getOpposite(), 8), getCombatMovementSpeed());
+    /**
+     * If the guard should stick close to the guard/patroll pos.
+     * @return true if so.
+     */
+    private boolean shouldStayCloseToPos()
+    {
+        return buildingGuards.getTask() == GuardTask.GUARD || (buildingGuards.getTask() == GuardTask.PATROL && buildingGuards.shallPatrolManually());
     }
 
     @Override
     public void guardMovement()
     {
+        if (worker.getRandom().nextInt(3) < 1)
+        {
+            worker.isWorkerAtSiteWithMove(buildingGuards.getGuardPos(), 3);
```

Comment:
what is this supposed to do?

Reply:
Atm, there is a problem that the archer, stays a very long time, far away from the actual guarding spot. This should balance out that regularly the archer returns to the initial guarding spot and then from there walk a little bit again.

Reply:
ok